src/Peering/Peering/Peering/UpdateAzurePeeringCommand.cs

```diff
@@ -0,0 +1,589 @@
+            ValueFromPipeline = true,
+            HelpMessage = Constants.PeeringObjectHelp,
+            ParameterSetName = Constants.ParameterSetNameDefault + Constants.Exchange,
+            DontShow = true)]
+        public PSPeering InputObject { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = Constants.ParameterSetNameByResourceId + Constants.Direct,
+            HelpMessage = Constants.ResourceIdHelp)]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = Constants.ParameterSetNameByResourceId + Constants.Exchange,
+            HelpMessage = Constants.ResourceIdHelp)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Gets or sets The Resource Group Name
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceGroupNameHelp,
+            ParameterSetName = Constants.Direct)]
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceGroupNameHelp,
+            ParameterSetName = Constants.Exchange)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets The InputObject NameMD5AuthenticationKeyHelp
+        /// </summary>
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.Direct)]
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.Exchange)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether use for peering service.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = Constants.UseForPeeringServiceHelp,
+            ParameterSetName = Constants.Direct)]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = Constants.ParameterSetNameDefault + Constants.Direct,
+            HelpMessage = Constants.UseForPeeringServiceHelp)]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = Constants.ParameterSetNameByResourceId + Constants.Direct,
+            HelpMessage = Constants.UseForPeeringServiceHelp)]
+        public bool? UseForPeeringService { get; set; }
+
+        /// <summary>
+        /// Gets or sets the exchange session.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            Position = 2,
+            ValueFromPipeline = true,
```

Comment:
@rileymckenna the `-InputObject` parameter is the only parameter in this cmdlet that should have `ValueFromPipeline`